pallets/loans/src/lib.rs

```diff
@@ -646,6 +645,63 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Propose a change.
+		/// The change is not performed until you call
+		/// [`Pallet::apply_loan_mutation()`].
+		#[pallet::weight(100_000_000)]
+		#[pallet::call_index(8)]
+		pub fn propose_loan_mutation(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			loan_id: T::LoanId,
+			mutation: LoanMutation<T::Rate>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			Self::ensure_role(pool_id, &who, PoolRole::LoanAdmin)?;
+
+			let (mut loan, _count) = Self::get_active_loan(pool_id, loan_id)?;
+			transactional::with_transaction(|| {
+				let result = loan.mutate_with(mutation.clone());
+
+				// We do not want to apply the mutation,
+				// only check if there is no error in applying it
+				TransactionOutcome::Rollback(result)
+			})?;
+
+			T::ChangeGuard::note(pool_id, Change::Loan(loan_id, mutation).into())?;
+
+			Ok(())
+		}
+
+		/// Apply a proposed change identified by a change id.
+		/// It will only perform the change if the requirements for it
+		/// are fulfilled.
+		#[pallet::weight(100_000_000)]
+		#[pallet::call_index(9)]
+		pub fn apply_loan_mutation(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			change_id: T::Hash,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let Change::Loan(loan_id, mutation) = Self::get_released_change(pool_id, change_id)? else {
+                Err(Error::<T>::UnrelatedChangeId)?
+			};
+
+			let (_, _count) = Self::update_active_loan(pool_id, loan_id, |loan| {
+				loan.mutate_with(mutation.clone())
+			})?;
+
+			Self::deposit_event(Event::<T>::Mutated {
+				pool_id,
+				loan_id,
+				mutation,
+			});
+
+			Ok(())
+		}
+
```

Comment:
This has not changed, I've only moved upwards in the file where they make more sense.

Not sure if we can also modify `call_index` of these calls to leave with the same order, or if it's too late. I think we could because by now nobody has used it. 
cc @NunoAlexandre @wischli

Reply:
@lemunozm considerations when changing pallets or call indexes:

1. Is there any governance proposal that encodes a call that will be changed?
2. Is there any dApp integration (our frontend included) that encodes a call to a pallet of any of our runtimes based on the pallet and call index?
3. Is there any data on storage that references a target pallet or call index?

If none apply, should be fine to change :) hope I didn't forget any edge case

Reply:
Thanks Nuno!

AFAIK, none of them apply here 🚀